src/ciflow-bot.ts

```diff
@@ -0,0 +1,220 @@
+import * as probot from 'probot';
+
+export default function runCIFlowBot(app: probot.Application): void {
+  new CIFlowBot(app).main();
+}
+
+export class CIFlowBot {
+  // Constructor required
+  readonly app: probot.Application;
+
+  // Configurations
+  readonly allowed_command: string[] = ['ciflow'];
+  readonly bot_app_name = 'pytorchbot';
+  readonly bot_assignee = 'pytorchbot';
+  readonly pr_label_prefix = 'ciflow/';
+  readonly rollout_users = ['zhouzhuojie']; // slow rollout to specific group of users first
+
+  // Stateful instance variables
+  command = '';
+  command_args: string[] = [];
+  comment_author = '';
+  comment_body = '';
+  ctx: probot.Context = null;
+  dispatch_labels: string[] = [];
+  dispatch_strategies: string[] = ['add_default_labels'];
+  event = '';
+  owner = '';
+  pr_author = '';
+  pr_labels: string[] = [];
+  pr_number = 0;
+  repo = '';
+
+  constructor(app: probot.Application) {
+    this.app = app;
+  }
+
+  valid(): boolean {
+    if (this.event !== 'pull_request' && this.event !== 'issue_comment') {
+      this.ctx.log.error({ctx: this.ctx}, 'Unknown webhook event');
+      return false;
+    }
+
+    if (this.event === 'issue_comment') {
+      if (this.comment_author === '') {
+        this.ctx.log.error({ctx: this.ctx}, 'Empty comment author');
+        return false;
+      }
+
+      // only the pr author can trigger new changes to the ciflow
+      if (this.comment_author !== this.pr_author) {
+        return false;
+      }
+
+      if (!this.allowed_command.includes(this.command)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  rollout(): boolean {
+    if (this.rollout_users.includes(this.pr_author)) {
+      return true;
+    }
+    return false;
+  }
+
+  async dispatch(): Promise<void> {
+    // Dispatch_strategies is like a pipeline of functions we can apply to
+    // change `this.dispatch_labels`. We can add other dispatch algorithms
+    // based on the ctx or user instructions.
+    // The future algorithms can manupulate the `this.dispatch_labels`, and
+    // individual workflows that can build up `if` conditions on the labels
+    // can be found in `.github/workflows` of pytorch/pytorch repo.
+    this.dispatch_strategies.map(this.dispatch_strategy_func.bind(this));
+
+    // Signal the dispatch to GitHub
+    await this.signal_github();
+
+    // Logging of the dispatch
+    this.ctx.log.info(
+      {
+        dispatch_labels: this.dispatch_labels,
+        dispatch_strategies: this.dispatch_strategies,
+        event: this.event,
+        owner: this.owner,
+        pr_number: this.pr_number,
+        pr_labels: this.pr_labels,
+        repo: this.repo
+      },
+      'ciflow dispatch success!'
+    );
+  }
+
+  dispatch_strategy_func(strategyName: string): void {
+    switch (strategyName) {
+      case 'add_default_labels':
```

Comment:
We should repeating string literals throughout the code
```suggestion
      case this.add_labels_distaptch_strategy:
```